lib/wickedbase.pm

```diff
@@ -1303,11 +1303,12 @@ sub need_network_tweaks() {
 }
 
 sub wait_for_background_process {
-    my ($self, $pid, %args) = @_;
-    $args{proceed_on_failure} //= 0;
+    my ($self, $pid) = @_;
+    # https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19853/files#r1718454521
 
-    my $ret = script_run("waitpid --timeout 25 $pid", %args);
-    if ($ret == 3) {    # timeout
+    enter_cmd "wait $pid; echo wait-DONE-\$?- | tee /dev/$serialdev";
+    my $ret = wait_serial "wait-DONE-\d+-";
```

Comment:
```suggestion
    my $ret = wait_serial "wait-DONE-\\d+-";
```
We need to mask the `\` so it is passed to wait_serial in the string...

Reply:
Let me use a tmp file to record the exit code, seems a bit simple logic :)

Reply:
New VRs:
[tw](https://openqa.opensuse.org/tests/4415222#dependencies)
[sle15sp6](https://openqa.suse.de/tests/15215087#dependencies)
[sle15sp4](https://openqa.suse.de/tests/15215089#dependencies)

Reply:
done, Thanks much @cfconrad :+1: 
[15sp4](https://openqa.suse.de/tests/15215558#dependencies)
[15sp6](https://openqa.suse.de/tests/15215556#dependencies)
[tw](https://openqa.opensuse.org/tests/4415408#dependencies)